packages/components/src/internal/components/user/SiteUsersGridPanel.tsx

```diff
@@ -113,97 +128,114 @@ export class SiteUsersGridPanel extends React.PureComponent<Props, State> {
+        );
+    }
+
     getUsersView(paramVal: string): string {
         return paramVal === 'inactive' || paramVal === 'all' ? paramVal : 'active'; // default to view active users
     }
 
-    getUsersModel(): QueryGridModel {
-        const { usersView } = this.state;
-        const gridId = 'user-management-users-' + usersView;
-        let baseFilters = List<Filter.IFilter>([Filter.create('active', usersView === 'active')]);
-
-        if (usersView === 'all') {
-            baseFilters = List<Filter.IFilter>();
-        }
-
-        const model = getStateQueryGridModel(gridId, SCHEMAS.CORE_TABLES.USERS, {
-            containerPath: '/',
-            omittedColumns: OMITTED_COLUMNS,
-            baseFilters,
-            bindURL: true,
-            isPaged: true,
-        });
+    getUsersModelId(): string {
+        return 'user-management-users-' + this.state.usersView;
+    }
 
-        return getQueryGridModel(model.getId()) || model;
+    getUsersModel(): QueryModel {
+        return this.props.queryModels[this.getUsersModelId()];
     }
 
-    toggleViewActive = (viewName: string) => {
-        replaceParameter(getLocation(), 'usersView', viewName);
+    toggleViewActive = (viewName: string): void => {
+        this.setState(() => ({ usersView: viewName }));
     };
 
-    toggleDialog = (name: string, requiresSelection = false) => {
-        if (requiresSelection && this.getUsersModel().selectedIds.size === 0) {
+    closeDialog = (): void => {
+        this.toggleDialog(undefined);
+    };
+
+    toggleDialog = (name: string, requiresSelection = false): void => {
+        if (requiresSelection && !this.getUsersModel().hasSelections) {
             this.setState(() => ({ showDialog: undefined }));
         } else {
             this.setState(() => ({ showDialog: name }));
         }
     };
 
-    onCreateComplete = (response: any, role: string) => {
-        this.toggleDialog(undefined); // close dialog
+    onCreateComplete = (response: any, role: string): void => {
+        this.closeDialog();
         this.onRowSelectionChange(this.getUsersModel(), undefined, false); // clear selected user details
         this.props.onCreateComplete(response, role);
+        this.reloadUsersModel();
     };
 
-    onUsersStateChangeComplete = (response: any, resetSelection = true) => {
-        this.toggleDialog(undefined); // close dialog
+    reloadUsersModel(): void {
+        this.props.actions.loadModel(this.getUsersModelId(), true);
+    }
+
+    onUsersStateChangeComplete = (response: any, resetSelection = true): void => {
+        this.closeDialog();
         if (resetSelection) {
             this.onRowSelectionChange(this.getUsersModel(), undefined, false); // clear selected user details
         }
-
         this.props.onUsersStateChangeComplete(response);
+        this.reloadUsersModel();
     };
 
-    onRowSelectionChange = (model, row, checked) => {
+    onRowSelectionChange = (model: QueryModel, row, checked): void => {
```

Comment:
Type for `row` and `checked` (though more obvious than `row`)?

Reply:
Done